src/client/sandbox/cookie/server-sync.js

```diff
@@ -0,0 +1,122 @@
+import nativeMethods from '../../sandbox/native-methods';
+import XhrSandbox from '../../sandbox/xhr';
+import settings from '../../settings';
+import { stringify as stringifyJSON } from '../../json';
+import { isIE9 } from '../../utils/browser';
+
+const SWITCH_BACK_TO_ASYNC_XHR_DELAY = 2000;
+
+export default class ServerSync {
+    constructor () {
+        this.useAsyncXhr = true;
+        this.queue       = [];
+        this.sendedQueue = null;
+        this.activeReq   = null;
+
+        // NOTE: When unloading, we should switch to synchronous XHR to be sure that we won't lose any cookies.
+        nativeMethods.windowAddEventListener.call(window, 'beforeunload', () => this._beforeUnloadHandler(), true);
+    }
+
+    static _createXMLHttpRequest (isAsync) {
+        var xhr = XhrSandbox.createNativeXHR();
+
+        xhr.open('POST', settings.get().cookieSyncUrl, isAsync);
+        xhr.setRequestHeader('Cache-Control', 'no-cache, no-store, must-revalidate');
+
+        return xhr;
+    }
+
+    static _isNoContentStatus (request) {
+        try {
+            // NOTE: XMLHTTPRequest implementation in MSXML HTTP does not handle HTTP responses
```

Comment:
https://stackoverflow.com/questions/10046972/msie-returns-status-code-of-1223-for-ajax-request